jerry-core/jrt/jrt.h

```diff
@@ -108,7 +113,11 @@ extern void __noreturn jerry_unreachable (const char *, const char *, const uint
     } \
   } while (0)
 
+#ifdef(__GNUC__)
 #define JERRY_UNREACHABLE() __builtin_unreachable ()
+#else /* !__GNUC__ */
+#define JERRY_UNREACHABLE() abort(); /* but what if they haven't included stdlib?!? */
```

Comment:
I have some memories about `__builtin_unreachable()`, when a bad code actually reached this point, and no return code was generated by the GCC. Hence the code entered into the next function. Hard to debug. This was the reason while we added an assert before `__builtin_unreachable()`, so we can capture these issues in debug mode.